pkgs/applications/networking/instant-messengers/telegram/kotatogram-desktop/tg_owt.nix

```diff
@@ -23,14 +23,26 @@ in stdenv.mkDerivation {
   nativeBuildInputs = [ pkg-config cmake ninja yasm ];
 
   buildInputs = [
-    libjpeg openssl libopus ffmpeg alsa-lib libpulseaudio protobuf
-    xorg.libX11 libXtst
+    libjpeg openssl libopus ffmpeg protobuf openh264 usrsctp libvpx abseil-cpp
+  ] ++ lib.optionals stdenv.isLinux [
+    libX11 libXtst libXcomposite libXdamage libXext libXrender libXrandr libXi
+    glib pipewire
   ];
 
   cmakeFlags = [
-    # Building as a shared library isn't officially supported and currently broken:
+    # Building as a shared library isn't officially supported and may break at any time.
     "-DBUILD_SHARED_LIBS=OFF"
+    # tdesktop has its own openal backend, it doesn't use backends from tg_owt
+    "-DTG_OWT_BUILD_AUDIO_BACKENDS=OFF"
+  ];
+
+  propagatedBuildInputs = [
+    # Required for linking downstream binaries.
+    abseil-cpp openh264 usrsctp libevent libvpx
   ];
 
-  meta.license = lib.licenses.bsd3;
+  meta = with lib; {
+    license = licenses.bsd3;
+    maintainers = with maintainers; [ ilya-fedin ];
```

Comment:
Missing description and homepage.

Reply:
It's a modified copy of tdesktop's tg_owt.nix, does it really need description?

Reply:
It could literally be ``modified version of tdesktop's tg_owt``

Reply:
tg_owt itself is not modified, it's a modified derivation [from tdesktop folder](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/networking/instant-messengers/telegram/tdesktop/tg_owt.nix), it differs only in used commit (since kotatogram is slower than tdesktop obviously) and patches for old macos sdk now (the latter I guess can be used by tdesktop's tg_owt derivation as well and tdesktop itself could be built for macos in nixpkgs)